java/lance-namespace-glue/src/main/java/com/lancedb/lance/namespace/glue/GlueNamespace.java

```diff
@@ -74,9 +99,349 @@ public ListNamespacesResponse listNamespaces(ListNamespacesRequest request) {
+              .catalogId(config.glueCatalogId())
+              .name(namespaceName)
+              .build());
+      return true;
+    } catch (EntityNotFoundException e) {
+      return false;
+    } catch (GlueException e) {
+      throw LanceNamespaceException.serverError(
+          "Failed to get Glue Database: " + namespaceName,
+          "/errors/glue-service-error",
+          instance,
+          formatGlueExceptionDetail(e));
+    }
+  }
+
+  private Database getDatabase(String namespaceName, String instance) {
+    try {
+      return glueClient
+          .getDatabase(
+              GetDatabaseRequest.builder()
+                  .catalogId(config.glueCatalogId())
+                  .name(namespaceName)
+                  .build())
+          .database();
+    } catch (EntityNotFoundException e) {
+      throw LanceNamespaceException.notFound(
+          "Glue database not found: " + namespaceName,
+          "/errors/namespace-not-found",
+          instance,
+          formatGlueExceptionDetail(e));
+    } catch (GlueException e) {
+      throw LanceNamespaceException.serverError(
+          "Failed to get Glue database: " + namespaceName,
+          "/errors/glue-service-error",
+          instance,
+          formatGlueExceptionDetail(e));
+    }
+  }
+
+  private void deleteDatabase(String namespaceName, String instance) {
+    try {
+      glueClient.deleteDatabase(
+          DeleteDatabaseRequest.builder()
+              .catalogId(config.glueCatalogId())
+              .name(namespaceName)
+              .build());
+    } catch (GlueException e) {
+      throw LanceNamespaceException.serverError(
+          "Failed to drop Glue database: " + namespaceName,
+          "/errors/glue-service-error",
+          instance,
+          formatGlueExceptionDetail(e));
+    }
+  }
+
+  private void deleteDatabaseOrSkip(
+      String namespaceName, String instance, DropNamespaceRequest.ModeEnum mode) {
+    try {
+      glueClient.deleteDatabase(
+          DeleteDatabaseRequest.builder()
+              .catalogId(config.glueCatalogId())
+              .name(namespaceName)
+              .build());
+    } catch (EntityNotFoundException e) {
+      if (mode != DropNamespaceRequest.ModeEnum.SKIP) {
+        throw LanceNamespaceException.notFound(
+            "Namespace not found: " + namespaceName,
+            "/errors/namespace-not-found",
+            instance,
+            formatGlueExceptionDetail(e));
+      }
+    } catch (GlueException e) {
+      throw LanceNamespaceException.serverError(
+          "Failed to drop Glue database: " + namespaceName,
+          "/errors/glue-service-error",
+          instance,
+          formatGlueExceptionDetail(e));
+    }
+  }
+
+  private DatabaseInput buildDatabaseInput(String namespaceName, Map<String, String> params) {
+    DatabaseInput.Builder builder = DatabaseInput.builder().name(namespaceName);
+
+    if (params.containsKey(GlueConstants.PARAM_LOCATION)) {
```

Comment:
I'm not familiar with glue, so I'm a little confused here. Please correct if I'm wrong.

If we don't set the location uri, will glue give it a default location uri? If glue database can have no location uri, then what's the behavior when creating a table under the database without specifying table location uri ?